File: Code Examples/Security_OTP_Example/src/main/java/io/hashimati/security/PasswordEncoderService.java
package io.hashimati.security;

import jakarta.inject.Singleton;
import org.jasypt.util.password.StrongPasswordEncryptor;

@Singleton
public class PasswordEncoderService implements PasswordEncoder{
    private StrongPasswordEncryptor strongPasswordEncryptor = new StrongPasswordEncryptor();
    @Override
    public String encode(String rawPassword) {
        return strongPasswordEncryptor.encryptPassword(rawPassword);
    }

    @Override
    public boolean matches(String rawPassword, String encodedPassword) {
        return strongPasswordEncryptor.checkPassword(rawPassword, encodedPassword);
    }

    @Override
    public boolean upgradeEncoding(String encodedPassword) {
        throw new RuntimeException("This function is not supported");
    }
}

File: Code Examples/JWT/Kotlin/JWTKotlinMongo/src/main/kotlin/io/hashimati/security/PasswordEncoderService.kt
package io.hashimati.security

import PasswordEncoder
import jakarta.inject.Singleton
import org.jasypt.util.password.StrongPasswordEncryptor

@Singleton
class PasswordEncoderService {
    private val strongPasswordEncryptor: StrongPasswordEncryptor = StrongPasswordEncryptor()
    fun encode(rawPassword: String?): String {
        return strongPasswordEncryptor.encryptPassword(rawPassword)
    }

    fun matches(rawPassword: String?, encodedPassword: String?): Boolean {
        return strongPasswordEncryptor.checkPassword(rawPassword, encodedPassword)
    }

    fun upgradeEncoding(encodedPassword: String?): Boolean {
        throw RuntimeException("This function is not supported")
    }
}